docs/rfcs/confmap-provider-configuration.md

```diff
@@ -0,0 +1,184 @@
+fragment.
+
+Advantages:
+
+- Not likely to be used by config backends for any of our supported protocols,
+  so has a low chance of conflict.
+- Fits into existing config URIs.
+
+Disadvantages:
+
+- Even if fragments are likely not useful to backends, we are still preventing
+  their use in upstream Providers.
+- Doesn't conform to the spirit of how fragments should be used according to RFC
+  3986.
+- Only allows easily specifying key-value pairs.
+
+We could likely partially circumvent the key-value pair limitation by
+recursively calling confmap Providers to resolve files, env vars, HTTP URLs,
+etc. For example:
+
+```text
+https://config.com/config#refresh-interval=env:REFRESH_INTERVAL&headers=file:headers.yaml
+```
+
+Using this strategy would also allow us to more easily get env vars and to get
+values from files for things like API tokens.
+
+### Add to the string passed to the config flag
+
+Instead of configuring Providers inside their URIs, we could add an extra
+section to the string passed to `--config` and configure how a URI is retrieved
+through that config. This approach is undesirable because the config flag
+currently only takes valid URIs, and updating the format to accommodate this
+would require we adopt an unconventional format.
+
+### Separate flags to configure Providers per config URI
+
+Advantages:
+
+- Allows us to keep config URIs opaque.
+- Options live right next to config URIs on the command line.
+- Does not require breaking changes.
+
+Disadvantages:
+
+- The flags would need to be placed in a certain position in the arguments list
+  to specify which URI they apply to.
+- Complicating the flags like this would be suboptimal UX.
+
+### Separate config file
+
+We could allow specifying a config file that contains all URIs to obtain the
+Collector's configuration from.
+
+Advantages:
+
+- Allows us to keep URIs opaque.
+- Map structures are easier to work with than command-line arguments for complex
+  config.
+- Shouldn't require any breaking changes.
+
+Disadvantages:
+
+- The Collector's main config file now depends on a separate config file, which
+  would be suboptimal UX.
+
+### Specify additional config sources inside the main Collector configuration
+
+This is a variant of providing a separate config source-only configuration file
+that instead puts those URIs and their options inside the main configuration
+file.
+
+Advantages:
+
+- Compared to the other option, keeps a single configuration file.
+- All advantages of the other file option.
+
+Disadvantages:
+
+- There are now two ways to include config inside a Collector config file.
+- Using the existing bracket syntax inside config files does not support
+  options, and neither does specifying them through the command line.
+- Complicates the config schema and the config resolution process.
+
+## Resolution
+
+We will configure providers through URI fragments. These are seldom used in
```

Comment:
We may want to explicitly exclude the `env` provider from here

Reply:
I called this out in an "Exceptions" section above, let me know if that covers it.